HeartbeatLogging/Tests/Unit/HeartbeatStorageTests.swift

```diff
@@ -0,0 +1,192 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import XCTest
+@testable import HeartbeatLogging
+
+class HeartbeatStorageTests: XCTestCase {
+  func testOfferHeartbeatThenFlush() throws {
+    // Given
+    let storage = HeartbeatStorage(id: #file, storage: StorageFake())
+    XCTAssertNil(storage.flush())
+    // When
+    storage.offer(Heartbeat(info: #function))
+    // Then
+    XCTAssertNotNil(storage.flush())
+  }
+
+  func testOfferHeartbeatWithStorageReadError() throws {
+    // Given
+    let (storageFake, queue) = (StorageFake(), DispatchQueue(label: #function))
+    let storage = HeartbeatStorage(id: #file, storage: storageFake, queue: queue)
+    XCTAssertNil(storage.flush())
+    storageFake.failOnNextRead = true
+    // When
+    storage.offer(Heartbeat(info: #function))
+    // Then
+    drain(queue)
+    // Expect
+    XCTAssertNotNil(storage.flush())
```

Comment:
TODO: validate heartbeat content. Here and below.

Reply:
Same as above comment. Added TODO